### ISSUE TITLE
Internal Dashboard: Enhancements

### ISSUE THREAD

**Issue by AkhileshNegi** [glific/glific#2790]
### Description

Reporting has been an integral part of Glific so NGOs can measure the impact of their program. Our current approach involves using BigQuery and LookerStudio for creating reports, where data is periodically synced to BigQuery, this involves syncing data to BigQuery at regular intervals, which can then serve as a data source for LookerStudio. As it takes additional effort to set up BigQuery and LookerStudio, NGOs expressed the need for some reporting right out of the box and accessible from within the platform.

This [issue](https://github.com/glific/glific/issues/2531) addresses the implementation of an internal dashboard that will complement, rather than replace, BigQuery and LookerStudio

We have deployed the first version of the internal dashboard and are now seeking to enhance it by incorporating additional graphs and metrics which will provide NGO partners to gain deeper insights from the dashboard. 

### Goals 

- Add the ability to export data from the dashboard 
- Periodically send data via mail to the organization(This can be based on daily, weekly, monthly or combination of any of these three) and organization can setup multiple email ids to whom this will be sent
- Replicate a few useful graphs from our LookerStudio template
- Add a few more graphs to get an overview of the organization's chatbot
  - #2909 
  - #2908 
  - #2907 
  - #2880 
  - #2879 
  - #2878
  - #2877 
  - #2876 
  - #2875
  - #2874
  - #2774

### Project Name:
Glific

### Organization Name: 
Project Tech4Dev

### Domain: 
Social Sector

### Complexity: 
Medium

### Tech Skills Needed:
Elixir, Liveview, Database

### Mentor(s): 
@AkhileshNegi 

### Category: 
Feature

### Product Set-Up: 
https://github.com/glific/

### Acceptance Criteria: 
PR merged into the main branch

**Comment by jairaizel** [comment]
Hi @AkhileshNegi and @mdshamoon I am highly interested in this project and worked with a similar dashboard project for EY Open Science Data Challenge, Rice Yield Prediction, where I placed 43rd Globally among several working data science professionals and students.
I also am extremely interested and looking forward to diving into the world of Gov Tech open source and creating a humongous impact for our country through this project as I am a mentor in the college's only Open source club where I led several students to contribute and participate in prestigious events such as GSSoC, TSoC, SSoC.
I would love to connect with you soon and work on some interesting first issues to start with. Let's embark on a great journey!!!

**Comment by RatneshKumarDubeyCodeExperiences** [comment]
**I am so much excited to participate in this project, i am a beginner and having zeal to make something better, under the guidance of respected Mentors i'll give my best in this project**

**Comment by KDwevedi** [comment]
comment

**Comment by aravind-glific** [comment]
**### Expected enhancements on Internal Dashboard :** 

1) Date range, Flow name filter for overall page 
2) Total number- Optin contacts , Optout contacts, On the bot but not optedin (if possible breakup as per language) 
3) Graph for Optin & Optout contacts 
4) Filter a flow name- Number of users (unique & total) who started the flow & completed (if possible) 
5) Total number of collections 
6) Short description of Monthly conversation - Contacts in a flow , % completion of each flow, most completed flows in a month, seen vs responded 
7) Place holder for (link+text) where NGOs can add their dashboard link for the teams 
8) Can we avoid re-login for internal dashboard (if possible?)
Make the internal dashboard configurable - see the contact variables data

**Comment by AkhileshNegi** [comment]
We have added all the metrics for the new version of the internal dashboard. We are working on the UI front now based on these designs
 https://www.figma.com/proto/hvzGlCIQBSwHQ70bsW5m9Z/Web-app?page-id=2%3A2649&type=design&node-id=1136-7481&viewport=-2362%2C-5866%2C0.25&t=AIt6f64S2CCjo4WV-1&scaling=min-zoom&mode=design
 Once this is done we'll release this and see partner NGOs adoption while making point of things we can add more for next version
